sklearn/metrics/_plot/precision_recall_curve.py

```diff
@@ -186,21 +310,42 @@ def plot(
         `drawstyle="default"`. However, the curve will not be strictly
         consistent with the reported average precision.
         """
-        self.ax_, self.figure_, name = self._validate_plot_params(ax=ax, name=name)
-
-        default_line_kwargs = {"drawstyle": "steps-post"}
-        if self.average_precision is not None and name is not None:
-            default_line_kwargs["label"] = (
-                f"{name} (AP = {self.average_precision:0.2f})"
+        names = _deprecate_singular(name, names, "name")
+        names_ = names if names else self.names
+        multi_args = [
+            input
+            for input in (self.average_precisions, names_)
+            if isinstance(input, list)
+        ]
+        multi_args.extend([self.precisions, self.recalls])
+        # I could make this message more informative (i.e. give
+        # lengths of each param) but it requires a fair bit more code
+        if len({len(arg) for arg in multi_args}) > 1:
+            raise ValueError(
+                "`self.precisions`, `self.recalls`, and if provided, "
+                "`self.average_precisions` and `self.names` (or `plot`'s `names`), "
+                "from `PrecisionRecallDisplay` initialization, should all "
+                "be lists of the same length."
             )
-        elif self.average_precision is not None:
-            default_line_kwargs["label"] = f"AP = {self.average_precision:0.2f}"
-        elif name is not None:
-            default_line_kwargs["label"] = name
 
-        line_kwargs = _validate_style_kwargs(default_line_kwargs, kwargs)
+        n_multi = len(self.precisions)
+        self.ax_, self.figure_, name = self._validate_plot_params(
+            ax=ax,
+            name=name,
+            n_multi=n_multi,
+            curve_type="PR",
+        )
+
+        line_kwargs = self._get_line_kwargs(n_multi, names_, fold_line_kws, **kwargs)
 
-        (self.line_,) = self.ax_.plot(self.recall, self.precision, **line_kwargs)
+        self.line_ = []
+        for recall, precision, line_kw in zip(
+            self.recalls, self.precisions, line_kwargs
+        ):
+            self.line_.extend(self.ax_.plot(recall, precision, **line_kw))
+        # Should we do this to be backwards compatible or have `line_` always be list?
+        if len(self.line_) == 1:
+            self.line_ = self.line_[0]
```

Comment:
Should `line_` always be a list or should we do this to be backwards compatible?